src/plots/heatmap2.ts

```diff
@@ -0,0 +1,206 @@
+import { Selection, BaseType, select } from 'd3-selection';
+import { axisBottom, axisLeft } from 'd3-axis';
+// import { line, curveLinearClosed } from 'd3-shape';
+
+import { scaleLinear, ScaleLinear } from 'd3-scale';
+
+import { uniqueId } from 'lodash-es';
+import { RGBColor, ColorMap, OpacityMap } from '@colormap/core';
+
+interface IMargins {
+  left: number;
+  bottom: number;
+  top: number;
+  right: number;
+}
+
+class HeatMap2 {
```

Comment:
We have HeadMap1? The plotly heat map, was there a reason this couldn't be resused?